editor/js/reaction.js

```diff
@@ -906,3 +907,23 @@ function enumToStrings(protoEnum) {
 function stringToEnum(name, protoEnum) {
   return protoEnum[name];
 }
+
+/**
+ * Switch the UI into a read-only mode. This is irreversible.
+ */
+function freeze() {
```

Comment:
How do you expect this will be called? Will the editor be loaded in a mutable state and then frozen? Ideally the user would never see the unfrozen state.

Reply:
I figured whoever is sorting out the search interface would just call freeze(). Does that make sense?

Before freeze: https://photos.app.goo.gl/aBsTius8ptqxK79k9

After freeze: https://photos.app.goo.gl/fjZZwV3fPrvnXwNi9

Reply:
Right, I'm wondering though if the page will load in a mutable state first? How quickly does the transition happen? Is it possible to load the form the first time in frozen mode?

Reply:
It used to be that the editor initialized the UI in a single JS task via the XHR response handler for the reaction request. Under that design, a call to freeze() at the end of that handler would set up the frozen DOM without any intermediate state getting rendered on the screen.

On another GitHub issue however, I recently read the troubling news that we have an initialization race. Even worse, we are solving it with a sleep.

If I understood that issue correctly, then I think all bets are off with regard to displaying intermediate states, and I would solve the problem you asked about by first understanding our initialization.

Reply:
Thanks for bringing up that issue; I'll merge this while we work on that.